app/helpers/search_helper.rb

```diff
@@ -1,4 +1,85 @@
 module SearchHelper
+  def search_posts
+    # Check permissions
+    posts = (current_user&.is_moderator || current_user&.is_admin ? Post : Post.undeleted)
+            .qa_only.list_includes
+
+    qualifiers = filters_to_qualifiers
+    search_string = params[:search]
+
+    # Filter based on search string qualifiers
+    if search_string.present?
+      search_data = parse_search(search_string)
+      qualifiers += parse_qualifier_strings search_data[:qualifiers]
+      search_string = search_data[:search]
+    end
+
+    posts = qualifiers_to_sql(qualifiers, posts)
+    posts = posts.paginate(page: params[:page], per_page: 25)
+
+    if search_string.present?
+      posts.search(search_data[:search]).user_sort({ term: params[:sort], default: :search_score },
+                                                   relevance: :search_score, score: :score, age: :created_at)
+    else
+      posts.user_sort({ term: params[:sort], default: :score },
+                      score: :score, age: :created_at)
+    end
+  end
+
+  # Convert a Filter record into a form parseable by the search function
+  def filter_to_qualifiers(filter)
+    qualifiers = []
+    qualifiers.append({ param: :score, operator: '>=', value: filter.min_score }) unless filter.min_score.nil?
+    qualifiers.append({ param: :score, operator: '<=', value: filter.max_score }) unless filter.max_score.nil?
+    qualifiers.append({ param: :answers, operator: '>=', value: filter.min_answers }) unless filter.min_answers.nil?
+    qualifiers.append({ param: :answers, operator: '<=', value: filter.max_answers }) unless filter.max_answers.nil?
+    qualifiers.append({ param: :include_tags, tag_ids: filter.include_tags }) unless filter.include_tags.nil?
+    qualifiers.append({ param: :exclude_tags, tag_ids: filter.exclude_tags }) unless filter.exclude_tags.nil?
+    qualifiers.append({ param: :status, value: filter.status }) unless filter.status.nil?
+    qualifiers
```

Comment:
consider (quality): is it better to iterate through the properties of `filter` and add to `qualifiers` for each that exists, rather than repeated `append/unless` statements?

Reply:
I'm not really sure how that would look to be honest; I'm not familiar enough with idiomatic Ruby patterns.

Reply:
Since the treatment of each property is different, I see no good way to do what you propose.